x-pack/solutions/search/plugins/search_connectors/server/services/index.ts

```diff
@@ -66,6 +80,7 @@ export class AgentlessConnectorsInfraService {
           id: connector.id,
           name: connector.name,
           service_type: connector.service_type,
+          is_deleted: connector.deleted,
```

Comment:
would use `!!connector.deleted`.

This handles cases where deleted might be null for newly created connectors by explicitly casting it to a boolean. Otherwise, you would need to account for this in your logic later on. JS is fun ...

Reply:
Makes sense :)